docker/release-images.mjs

```diff
@@ -98,118 +99,186 @@ import { execSync } from "node:child_process";
-  parsedArgs[arg] = argValue;
+// This is kinda gross but means we can just run the script directly and it will
+// work without running pnpm install before hand...
+async function lazyImports() {
+  await import("zx/globals");
+  toml = await import("toml");
+  core = await import("@actions/core");
 }
 
-let crane;
+const Environment = {
+  CI: "ci",
+  LOCAL: "local",
+  TEST: "test",
+};
 
-if (process.env.CI === "true") {
-  console.log("installing crane automatically in CI");
-  await $`curl -sL https://github.com/google/go-containerregistry/releases/download/v0.15.1/go-containerregistry_Linux_x86_64.tar.gz > crane.tar.gz`;
-  const sha = (await $`shasum -a 256 ./crane.tar.gz | awk '{ print $1 }'`).toString().trim();
-  if (sha !== "d4710014a3bd135eb1d4a9142f509cfd61d2be242e5f5785788e404448a4f3f2") {
-    console.error(chalk.red(`ERROR: sha256 mismatch for crane.tar.gz got: ${sha}`));
-    process.exit(1);
+
+function getEnvironment() {
+  if (process.env.CI === "true") {
+    return Environment.CI;
+  } else if (import.meta.jest !== undefined) {
+    return Environment.TEST;
+  } else {
+    return Environment.LOCAL;
   }
-  await $`tar -xf crane.tar.gz`;
-  crane = "./crane";
-} else {
-  if ((await $`command -v crane`.exitCode) !== 0) {
-    console.log(
-      chalk.red(
-        "ERROR: could not find crane binary in PATH - follow https://github.com/google/go-containerregistry/tree/main/cmd/crane#installation to install",
-      ),
-    );
-    process.exit(1);
+}
+
+function reportError(message, opts={throwOnFailure: false}) {
+  if (getEnvironment() === Environment.CI) {
+    core.setFailed(message);
+  } else if (getEnvironment() === Environment.LOCAL) {
+    console.error(message);
+  } else if (getEnvironment() === Environment.TEST) {
+    // Errors in tests are expected and mess up formatting
+    console.log(message);
+  }
+  if (opts.throwOnFailure) {
+    throw new Error(message);
   }
-  crane = "crane";
 }
 
-const AWS_ECR = `${parsedArgs.AWS_ACCOUNT_ID}.dkr.ecr.us-west-2.amazonaws.com/aptos`;
-const GCP_ARTIFACT_REPO = parsedArgs.GCP_DOCKER_ARTIFACT_REPO;
-const GCP_ARTIFACT_REPO_US = parsedArgs.GCP_DOCKER_ARTIFACT_REPO_US;
-const DOCKERHUB = "docker.io/aptoslabs";
+async function main() {
+  const REQUIRED_ARGS = ["GIT_SHA", "GCP_DOCKER_ARTIFACT_REPO", "GCP_DOCKER_ARTIFACT_REPO_US", "AWS_ACCOUNT_ID", "IMAGE_TAG_PREFIX"];
+  const OPTIONAL_ARGS = ["WAIT_FOR_IMAGE_SECONDS", "DRY_RUN"];
 
-const INTERNAL_TARGET_REGISTRIES = [
-  GCP_ARTIFACT_REPO,
-  GCP_ARTIFACT_REPO_US,
-  AWS_ECR,
-];
+  const parsedArgs = {};
 
-const ALL_TARGET_REGISTRIES = [
-  ...INTERNAL_TARGET_REGISTRIES,
-  DOCKERHUB,
-];
+  await assertExecutingInRepoRoot();
 
-// default 10 seconds
-parsedArgs.WAIT_FOR_IMAGE_SECONDS = parseInt(parsedArgs.WAIT_FOR_IMAGE_SECONDS ?? 10, 10);
+  for (const arg of REQUIRED_ARGS) {
+    const argValue = argv[arg.toLowerCase().replaceAll("_", "-")] ?? process.env[arg];
```

Comment:
maybe split this into another function?

Reply:
seems unnecessary, also this was not my code I just moved it into a main function